core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -867,16 +867,23 @@ private long listBlobsToGetLatestIndexId() throws IOException {
     }
 
     private void writeAtomic(final String blobName, final BytesReference bytesRef) throws IOException {
-        final String tempBlobName = "pending-" + blobName;
+        final String tempBlobName = "pending-" + UUIDs.randomBase64UUID() + "-" + blobName;
         try (InputStream stream = bytesRef.streamInput()) {
             snapshotsBlobContainer.writeBlob(tempBlobName, stream, bytesRef.length());
-        }
-        try {
             snapshotsBlobContainer.move(tempBlobName, blobName);
         } catch (IOException ex) {
-            // Move failed - try cleaning up
-            snapshotsBlobContainer.deleteBlob(tempBlobName);
-            throw ex;
+            IOException exceptionToThrow = ex;
+            try {
+                snapshotsBlobContainer.deleteBlob(tempBlobName);
+            } catch (NoSuchFileException e) {
```

Comment:
I don't think we should special-case this. Depending on underlying blobcontainer, a different exception might be thrown.